cfme/containers/image.py

```diff
@@ -152,6 +153,81 @@ def all(self):
                                            provider=provider or get_crud_by_name(ems_name)))
         return images
 
+    def check_compliance_multiple_images(self, image_entitys, wait_for_finish=True, timeout=240):
+        """Initiates compliance check and waits for it to finish."""
+
+        # Chose Check Compliance of Last Known Configuration
+        images_view = navigate_to(self, 'All')
+        self.check_image_entitys(image_entitys)
+        wait_for(lambda: images_view.toolbar.policy.is_enabled, num_sec=5,
+                 message='Policy drop down menu is disabled after checking some Images')
+        images_view.toolbar.policy.item_select('Check Compliance of Last Known Configuration',
+                                  handle_alert=True)
+        images_view.flash.assert_no_error()
+
+        # Verify Image summary
+        for image_instance in image_entitys:
+            original_state = 'Never Verified'
+            if wait_for_finish:
+                try:
+                    wait_for(
+                        lambda: image_instance.compliance_status != original_state,
+                        num_sec=timeout, delay=5,
+                        message='compliance state of Image ID, "{}", still matches {}'
+                                .format(image_instance.id, original_state)
+                    )
+                except TimedOutError:
+                    logger.error('compliance state of Image ID, "{}", is {}'
+                                 .format(image_instance.id, image_instance.compliance_status))
+                    raise TimedOutError('Timeout exceeded, Waited too much'
+                                        ' time for check Compliance finish ({}).'.format(timeout))
+
+    def check_image_entitys(self, image_entitys):
+        """check rows on Container Images table."""
+        images_view = navigate_to(self, 'All', use_resetter=False)
+        images_view.paginator.set_items_per_page(1000)
+        for image_entity in image_entitys:
+            images_view.entities.get_entity(name=image_entity.name, id=image_entity.id).check()
+
+    def perform_smartstate_analysis_multiple_images(
+            self, image_entitys, wait_for_finish=False, timeout='20M'):
+        """Performing SmartState Analysis on this Image
+        """
+
+        # task_name change from str to regular expression
+        # the str compile on tasks module
+        task_name = '(?i)(Container Image.*)'
+        task_type = 'container'
+        num_of_tasks = len(image_entitys)
+        images_view = navigate_to(self, 'All')
+        self.check_image_entitys(image_entitys)
+
+        images_view.toolbar.configuration.item_select(
+            'Perform SmartState Analysis', handle_alert=True)
+        for image_entity in image_entitys:
+            assert filter(lambda m: '"{}": Analysis successfully initiated'
```

Comment:
__REQUIRED__
Please do:
```python
for image_entity in image_entities:
    assert images_view.flash.assert_success_message(
        '"{}": Analysis successfully initiated'.format(image_entity.name), partial=True
    )
```
Instead.

Reply:
Done

Reply:
My apologies, remove the assert before flash.assert_success_message.